analyzers/src/SonarAnalyzer.CSharp/Rules/CryptographicKeyShouldNotBeTooShort.cs

```diff
@@ -98,17 +100,17 @@ protected override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var objectCreation = (ObjectCreationExpressionSyntax)c.Node;
-                    if (!(c.SemanticModel.GetTypeInfo(objectCreation).Type is ITypeSymbol containingType))
+                    var objectCreation = ObjectCreationFactory.Create(c.Node);
+                    if (!(c.SemanticModel.GetTypeInfo(objectCreation.Expression).Type is ITypeSymbol containingType))
```

Comment:
Let's refactor this old code to positive condition with actions in if block.
The `!condition` with `containingType` sideeffect and `return` is a double indirection (hard to read)